podvm: failing to build libseccomp when ARCH is not passed

The `Dockerfile.podvm` dockerfile accepts the `ARCH` argument which on its absence has caused a fail on the libseccomp build.

Notice on the logs shown below the message `Makefile.inc:27: A foreign ARCH was passed, but no CC alternative. Using -linux-musl-gcc as best guess` that indicates ARCH is set, which is not the case. Thus, we should ensure ARCH is an empty string by default.

```
$ docker build --no-cache -t podvm_libvirt --build-arg BUILDER_IMG=podvm_builder:latest         --build-arg CLOUD_PROVIDER=libvirt -f Dockerfile.podvm .
Sending build context to Docker daemon  704.1MB
Step 1/19 : ARG BUILDER_IMG="quay.io/confidential-containers/podvm-builder-ubuntu"
Step 2/19 : FROM ${BUILDER_IMG} AS podvm_builder
 ---> a95cea55fbf0
Step 3/19 : ARG CLOUD_PROVIDER=libvirt
 ---> Running in 167df7f6564b
Removing intermediate container 167df7f6564b
 ---> d890cc281e7f
Step 4/19 : ARG PODVM_DISTRO=ubuntu
 ---> Running in 4b4dcc989d3c
Removing intermediate container 4b4dcc989d3c
 ---> 134157164b8a
Step 5/19 : ARG AA_KBC=offline_fs_kbc
 ---> Running in 82f756633c60
Removing intermediate container 82f756633c60
 ---> b140f53f117d
Step 6/19 : ARG ARCH
 ---> Running in 358a723bd7bb
Removing intermediate container 358a723bd7bb
 ---> b31e548e7002
Step 7/19 : ENV CLOUD_PROVIDER ${CLOUD_PROVIDER}
 ---> Running in b7e0be6bfc06
Removing intermediate container b7e0be6bfc06
 ---> 3e4839705243
Step 8/19 : ENV PODVM_DISTRO ${PODVM_DISTRO}
 ---> Running in c01bb5d74eb9
Removing intermediate container c01bb5d74eb9
 ---> 5265c4c14f63
Step 9/19 : ENV AA_KBC ${AA_KBC}
 ---> Running in 415197bcb6a4
Removing intermediate container 415197bcb6a4
 ---> c70a22199bc5
Step 10/19 : ENV ARCH ${ARCH}
 ---> Running in 4546a599f790
Removing intermediate container 4546a599f790
 ---> 45b2354f20f9
Step 11/19 : RUN ./cloud-api-adaptor/podvm/hack/cross-build-extras.sh
 ---> Running in 0a7bcfab2c63
Removing intermediate container 0a7bcfab2c63
 ---> 26b168f4597e
Step 12/19 : ARG UBUNTU_IMAGE_URL=https://cloud-images.ubuntu.com/releases/focal/release-20230107/ubuntu-20.04-server-cloudimg-amd64.img
 ---> Running in e98f5a362f27
Removing intermediate container e98f5a362f27
 ---> d57e81483314
Step 13/19 : ARG UBUNTU_IMAGE_CHECKSUM=3895e38566e5c2c019f5c6f825ab7570ee34dac6b9142fab0c7e5a78084c4280
 ---> Running in e8cebc081be7
Removing intermediate container e8cebc081be7
 ---> 52703ba02a83
Step 14/19 : ENV UBUNTU_IMAGE_URL ${UBUNTU_IMAGE_URL}
 ---> Running in 4c194db578b0
Removing intermediate container 4c194db578b0
 ---> 2908dcdd3251
Step 15/19 : ENV UBUNTU_IMAGE_CHECKSUM ${UBUNTU_IMAGE_CHECKSUM}
 ---> Running in ca2ecf1bf44d
Removing intermediate container ca2ecf1bf44d
 ---> 3dc478ed7794
Step 16/19 : RUN cd cloud-api-adaptor/podvm &&      make binaries
 ---> Running in 6e1cdad7159c
Makefile.inc:27: A foreign ARCH was passed, but no CC alternative. Using -linux-musl-gcc as best guess
cd "../" && ARCH= make agent-protocol-forwarder
make[1]: Entering directory '/src/cloud-api-adaptor'
go: downloading github.com/containerd/ttrpc v1.1.0
go: downloading github.com/kata-containers/kata-containers/src/runtime v0.0.0-20220913141151-9b49a6ddc6fd
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/vishvananda/netlink v1.1.1-0.20220115184804-dd687eb2f2d4
go: downloading golang.org/x/sys v0.0.0-20220722155257-8c9f86f7a55f
go: downloading github.com/opencontainers/runtime-spec v1.0.3-0.20211214071223-8958f93039ab
go: downloading github.com/coreos/go-iptables v0.6.0
go: downloading github.com/containerd/containerd v1.6.6
go: downloading github.com/containernetworking/plugins v1.1.1
go: downloading github.com/vishvananda/netns v0.0.0-20210104183010-2eb08e3e575f
go: downloading github.com/vmware/govmomi v0.29.0
go: downloading github.com/avast/retry-go/v4 v4.3.2
go: downloading github.com/containers/podman/v4 v4.2.0
go: downloading google.golang.org/grpc v1.47.0
go: downloading k8s.io/cri-api v0.23.1
go: downloading github.com/sirupsen/logrus v1.9.0
go: downloading google.golang.org/genproto v0.0.0-20180817151627-c66870c02cf8
go: downloading github.com/golang/protobuf v1.5.2
go: downloading golang.org/x/net v0.0.0-20220722155237-a158d28d115b
go: downloading google.golang.org/protobuf v1.28.1
go: downloading golang.org/x/text v0.3.7
GOOS=linux GOARCH= go build -tags=libvirt -o "agent-protocol-forwarder" "cmd/agent-protocol-forwarder/main.go"
make[1]: Leaving directory '/src/cloud-api-adaptor'
install -D --compare "..//agent-protocol-forwarder" "/src/cloud-api-adaptor/podvm/files/usr/local/bin/agent-protocol-forwarder"
CC=-linux-musl-gcc ARCH= ../../kata-containers/ci/install_libseccomp.sh /src/cloud-api-adaptor/podvm/staticlib/kata-libseccomp /src/cloud-api-adaptor/podvm/staticlib/kata-gperf
/src/src/github.com/kata-containers/tests /src/cloud-api-adaptor/podvm
Already on 'main'
Your branch is up to date with 'origin/main'.
/src/cloud-api-adaptor/podvm
Already on 'main'
Your branch is up to date with 'origin/main'.
Installed yq version 3.4.1
/tmp/build-libseccomp.6lrK1 /src/cloud-api-adaptor/podvm
Build and install gperf version 3.1
/tmp/build-libseccomp.6lrK1/gperf-3.1 /tmp/build-libseccomp.6lrK1 /src/cloud-api-adaptor/podvm
checking whether make sets $(MAKE)... yes
configure: creating ./config.status
config.status: creating Makefile
=== configuring in lib (/tmp/build-libseccomp.6lrK1/gperf-3.1/lib)
configure: running /bin/bash ./configure --disable-option-checking '--prefix=/src/cloud-api-adaptor/podvm/staticlib/kata-gperf'  --cache-file=/dev/null --srcdir=.
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
checking for ar... ar
checking for ranlib... ranlib
checking for a BSD compatible install... /usr/bin/install -c
configure: creating ./config.status
config.status: creating Makefile
=== configuring in src (/tmp/build-libseccomp.6lrK1/gperf-3.1/src)
configure: running /bin/bash ./configure --disable-option-checking '--prefix=/src/cloud-api-adaptor/podvm/staticlib/kata-gperf'  --cache-file=/dev/null --srcdir=.
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
checking for a BSD compatible install... /usr/bin/install -c
checking for stack-allocated variable-size arrays... yes
checking for rand in -lm... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating config.h
=== configuring in tests (/tmp/build-libseccomp.6lrK1/gperf-3.1/tests)
configure: running /bin/bash ./configure --disable-option-checking '--prefix=/src/cloud-api-adaptor/podvm/staticlib/kata-gperf'  --cache-file=/dev/null --srcdir=.
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
configure: creating ./config.status
config.status: creating Makefile
=== configuring in doc (/tmp/build-libseccomp.6lrK1/gperf-3.1/doc)
configure: running /bin/bash ./configure --disable-option-checking '--prefix=/src/cloud-api-adaptor/podvm/staticlib/kata-gperf'  --cache-file=/dev/null --srcdir=.
checking whether make sets $(MAKE)... yes
checking for a BSD compatible install... /usr/bin/install -c
configure: creating ./config.status
config.status: creating Makefile
make[1]: Entering directory '/tmp/build-libseccomp.6lrK1/gperf-3.1'
cd lib; make all
make[2]: Entering directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/lib'
gcc -g -O2 -I. -c ./getopt.c
gcc -g -O2 -I. -c ./getopt1.c
g++ -g -O2 -I. -c ./getline.cc
g++ -g -O2 -I. -c ./hash.cc
rm -f libgp.a
ar rc libgp.a getopt.o getopt1.o getline.o hash.o
ranlib libgp.a
make[2]: Leaving directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/lib'
cd src; make all
make[2]: Entering directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/src'
g++ -g -O2 -I. -I./../lib -c ./version.cc
g++ -g -O2 -I. -I./../lib -c ./positions.cc
g++ -g -O2 -I. -I./../lib -c ./options.cc
g++ -g -O2 -I. -I./../lib -c ./keyword.cc
g++ -g -O2 -I. -I./../lib -c ./keyword-list.cc
g++ -g -O2 -I. -I./../lib -c ./input.cc
g++ -g -O2 -I. -I./../lib -c ./bool-array.cc
g++ -g -O2 -I. -I./../lib -c ./hash-table.cc
g++ -g -O2 -I. -I./../lib -c ./search.cc
g++ -g -O2 -I. -I./../lib -c ./output.cc
g++ -g -O2 -I. -I./../lib -c ./main.cc
g++ -g -O2  -o gperf version.o positions.o options.o keyword.o keyword-list.o input.o bool-array.o hash-table.o search.o output.o main.o ../lib/libgp.a -lm
make[2]: Leaving directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/src'
cd tests; make all
make[2]: Entering directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/tests'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/tests'
cd doc; make all
make[2]: Entering directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/doc'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/doc'
make[1]: Leaving directory '/tmp/build-libseccomp.6lrK1/gperf-3.1'
make[1]: Entering directory '/tmp/build-libseccomp.6lrK1/gperf-3.1'
cd lib; make install
make[2]: Entering directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/lib'
make[2]: Nothing to be done for 'install'.
make[2]: Leaving directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/lib'
cd src; make install
make[2]: Entering directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/src'
/bin/sh ./../build-aux/mkinstalldirs /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/bin
mkdir -p -- /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/bin
/usr/bin/install -c gperf /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/bin/gperf
make[2]: Leaving directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/src'
cd tests; make install
make[2]: Entering directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/tests'
make[2]: Nothing to be done for 'install'.
make[2]: Leaving directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/tests'
cd doc; make install
make[2]: Entering directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/doc'
/bin/sh ./../build-aux/mkinstalldirs /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/share/info
mkdir -p -- /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/share/info
/usr/bin/install -c -m 644 ./gperf.info /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/share/info/gperf.info
/bin/sh ./../build-aux/mkinstalldirs /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/share/man/man1
mkdir -p -- /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/share/man/man1
/usr/bin/install -c -m 644 ./gperf.1 /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/share/man/man1/gperf.1
/bin/sh ./../build-aux/mkinstalldirs /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/share/doc/
mkdir -p -- /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/share/doc/
/usr/bin/install -c -m 644 ./gperf.html /src/cloud-api-adaptor/podvm/staticlib/kata-gperf/share/doc//gperf.html
make[2]: Leaving directory '/tmp/build-libseccomp.6lrK1/gperf-3.1/doc'
make[1]: Leaving directory '/tmp/build-libseccomp.6lrK1/gperf-3.1'
/tmp/build-libseccomp.6lrK1 /src/cloud-api-adaptor/podvm
Gperf installed successfully
Build and install libseccomp version 2.5.4
/tmp/build-libseccomp.6lrK1/libseccomp-2.5.4 /tmp/build-libseccomp.6lrK1 /src/cloud-api-adaptor/podvm
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for x86_64-strip... no
checking for strip... strip
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking how to create a pax tar archive... gnutar
checking for x86_64-gcc... -linux-musl-gcc
checking whether the C compiler works... no
configure: error: in `/tmp/build-libseccomp.6lrK1/libseccomp-2.5.4':
configure: error: C compiler cannot create executables
See `config.log' for more details
make: *** [Makefile.inc:100: /src/cloud-api-adaptor/podvm/staticlib/kata-libseccomp/lib/libseccomp.a] Error 77
The command '/bin/sh -c cd cloud-api-adaptor/podvm &&      make binaries' returned a non-zero code: 2
```

--- a/podvm/Dockerfile.podvm
+++ b/podvm/Dockerfile.podvm
@@ -12,7 +12,7 @@ ARG CLOUD_PROVIDER=libvirt
 ARG PODVM_DISTRO=ubuntu
 ARG AA_KBC=offline_fs_kbc
 # If not provided, uses system architecture
-ARG ARCH
+ARG ARCH=x86_64
 
 ENV CLOUD_PROVIDER ${CLOUD_PROVIDER}
 ENV PODVM_DISTRO ${PODVM_DISTRO}

